linkerd/app/inbound/src/policy/http.rs

```diff
@@ -90,32 +110,90 @@ where
         task::Poll::Ready(Ok(()))
     }
 
-    fn call(&mut self, req: Req) -> Self::Future {
-        tracing::trace!(policy = ?self.policy, "Authorizing request");
-        match self.policy.check_authorized(self.client_addr, &self.tls) {
-            Ok(permit) => {
-                tracing::debug!(
-                    ?permit,
-                    tls = ?self.tls,
-                    client = %self.client_addr,
-                    "Request authorized",
-                );
-                self.metrics.allow(&permit, self.tls.clone());
-                let svc = self.inner.new_service((permit, self.target.clone()));
-                future::Either::Left(svc.oneshot(req).err_into::<Error>())
-            }
-            Err(e) => {
+    fn call(&mut self, req: ::http::Request<B>) -> Self::Future {
+        let server = self.policy.server.borrow();
+        let labels = ServerLabel(server.meta.clone());
+
+        let labels = RouteLabels {
+            route: self.default_route_meta.clone(),
+            server: labels,
+        };
+
+        let permit = match Self::check_authorized(
+            &*server.authorizations,
+            &self.meta,
+            labels,
+            &self.metrics,
+        ) {
```

Comment:
unimportant style nit: is there a reason `check_authorized` can't just take `&self`, rather than having to call it with `&self.meta` and `&self.metrics`?  or is `self` mutably borrowed somewhere i'm missing?